localstack/services/kms/provider.py

```diff
@@ -1018,6 +1019,33 @@ def _validate_grant_request(self, data: Dict, store: KmsStore):
                     f" constraint: [Member must satisfy enum value set: {VALID_OPERATIONS}]"
                 )
 
+    def _validate_plaintext_key_type_based(
+        self,
+        plaintext: PlaintextType,
+        key: KmsKey,
+        encryption_algorithm: EncryptionAlgorithmSpec = None,
+    ):
+        # max size values extracted from AWS boto3 documentation
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/encrypt.html
+        max_size_bytes = 4096  # max allowed size
+        if key.metadata["KeySpec"] == "RSA_2048" and encryption_algorithm == "RSAES_OAEP_SHA_1":
```

Comment:
We can use the key spec values from https://github.com/localstack/localstack/blob/ed7e02f18253e22780131cf961304aa9e9df5c85/localstack/aws/api/kms/__init__.py#L158 and for encryption algorithm https://github.com/localstack/localstack/blob/ed7e02f18253e22780131cf961304aa9e9df5c85/localstack/aws/api/kms/__init__.py#L122.